File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Conne/CategoryForm.ts
import { StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface CategoryForm {
    Name: StringEditor;
    Description: StringEditor;
    Photo: StringEditor;
}

export class CategoryForm extends PrefixedContext {
    static readonly formKey = 'Conne.Category';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoryForm.init)  {
            CategoryForm.init = true;

            var w0 = StringEditor;

            initFormType(CategoryForm, [
                'Name', w0,
                'Description', w0,
                'Photo', w0
            ]);
        }
    }
}

File: AdminPanel/AdminPanel.Web/Modules/Conne/Category/CategoryForm.cs
using Serenity.ComponentModel;

namespace AdminPanel.Conne.Forms;

[FormScript("Conne.Category")]
[BasedOnRow(typeof(CategoryRow), CheckNames = true)]
public class CategoryForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Photo { get; set; }
}